packages/curve-ui-kit/src/shared/ui/DataTable/DataTable.tsx

```diff
@@ -46,19 +51,22 @@ export const DataTable = <T extends TableItem>({
   const top = useLayoutStore((state) => state.navHeight)
   useScrollToTopOnFilterChange(table)
 
-  return (
+  const TableComponent = (
     <Table
       sx={{
         backgroundColor: (t) => t.design.Layer[1].Fill,
         borderCollapse: 'separate' /* Don't collapse to avoid funky stuff with the sticky header */,
+        ...(options?.maxHeight && {
+          maxHeight: options.maxHeight,
+          overflowY: 'auto',
+        }),
       }}
       data-testid={!loading && 'data-table'}
     >
       <TableHead
         sx={(t) => ({
           zIndex: t.zIndex.tableHeader,
-          position: 'sticky',
-          top,
+          ...(options?.disableStickyHeader ? { position: 'static' } : { position: 'sticky', top }),
```

Comment:
`...(!options?.disableStickyHeader && { position: 'sticky', top }),`
Unless somehow `{ position: 'static' }` is absolutely needed?

Reply:
zIndex is also only needed for stickyness